addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1155,6 +1155,280 @@ registry.anchorSlide = publicWidget.Widget.extend({
+    },
+    /**
+     * @override
+     */
+    destroy: function () {
+        clearInterval(this.setInterval);
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Draws the whole countdown, including one countdown for each time unit.
+     *
+     * @private
+     */
+    _drawCountdown: function () {
+        this._updateTimeDiff();
+        _.each(this.diff, (val, index) => {
+            var canvas = val['canvas'];
+            var ctx = canvas.getContext("2d");
+            ctx.canvas.width = this.size;
+            ctx.canvas.height = this.size;
+            this._clearCanvas(ctx);
+
+            $(canvas).toggleClass('d-none', !val['visible']);
+            if (!val['visible']) {
+                return;
+            }
+
+            if (this.layout === 'dark') {
+                this._drawBgCircle(ctx, true);
+            }
+            if (this.layout === 'thin-filled') {
+                this._drawBgCircle(ctx);
+            }
+            this._drawText(canvas, val['nb'], index);
+            if (this.layout === 'light') {
+                this._drawOuterCircleBg(ctx);
+            }
+            this._drawOuterCicle(ctx, val['nb'], val['total'], this.layout === 'thin' || this.layout === 'thin-filled');
+        });
+    },
+    /**
+     * Initializes the `diff` object. It will contains for every time unit the
+     * related canvas, its total step and its visibility.
+     *
+     * @private
+     */
+    _initTimeDiff: function () {
+        this.diff = {
+            'days': {
+                'canvas': this.$el.find('.s_countdown_days canvas')[0],
+                'total': 50, // TODO RATIO DAYS ? Based on how many days ?
+                'visible': this._isUnitVisible('d'),
+            },
+            'hours': {
+                'canvas': this.$el.find('.s_countdown_hours canvas')[0],
+                'total': 24,
+                'visible': this._isUnitVisible('h'),
+            },
+            'minutes': {
+                'canvas': this.$el.find('.s_countdown_minutes canvas')[0],
+                'total': 60,
+                'visible': this._isUnitVisible('m'),
+            },
+            'seconds': {
+                'canvas': this.$el.find('.s_countdown_seconds canvas')[0],
+                'total': 60,
+                'visible': this._isUnitVisible('s'),
+            },
+        };
+    },
+    /**
+     * Manages the countdown by drawing it and handling the end action.
+     * Call every seconds by the setInterval.
+     *
+     * @private
+     */
+    _manageCountdown: function () {
+        this._drawCountdown();
+        if (this.isFinished) {
+            clearInterval(this.setInterval);
+            if (!this.editableMode) {
+                this._handleEndCountdown();
+            }
+        }
```

Comment:
Not sure why this is not all in the _drawCountdown method ?

Reply:
I am not sure to understand, is not it a simple check to see if it is the first render or not ?

Reply:
Indeed using a variable to store the fact the user was on the countdown before it ends cleans the code and is better overall.
I thought about it but did not like the principle to store that info and preferred to split the drawing and countdown end handling.
Done